cmd/init.go

```diff
@@ -257,146 +274,152 @@ func (cmd *InitCmd) Run(cobraCmd *cobra.Command, args []string) {
-		if err != nil {
-			log.Fatal(err)
-		}
+func (cmd *InitCmd) getRegistryURL(cloudRegistryHostname string) string {
+	useDockerHub := "Use " + dockerHubHostname
+	useDevSpaceRegistry := "Use " + cloudRegistryHostname + " (free, private Docker registry)"
+	useOtherRegistry := "Use other registry"
+	registryUsernameHint := " => you are logged in as %s"
+	registryDefaultOption := useDevSpaceRegistry
+	registryLoginHint := "Please login via `docker login%s` and try again."
 
-		// Create generated yaml if cloud
-		generatedConfig, err := generated.LoadConfig()
-		if err != nil {
-			log.Fatal(err)
+	config := configutil.GetConfig()
+	dockerClient, err := docker.NewClient(config, false, log.GetInstance())
+	if err == nil {
+		authConfig, err := docker.GetAuthConfig(dockerClient, dockerHubHostname, true)
+		if err == nil && authConfig.Username != "" {
+			useDockerHub = useDockerHub + fmt.Sprintf(registryUsernameHint, authConfig.Username)
+			registryDefaultOption = useDockerHub
 		}
 
-		generatedConfig.CloudSpace = &generated.CloudSpaceConfig{
-			ProviderName: *cmd.providerName,
+		authConfig, err = docker.GetAuthConfig(dockerClient, cloudRegistryHostname, true)
+		if err == nil && authConfig.Username != "" {
+			useDevSpaceRegistry = useDevSpaceRegistry + fmt.Sprintf(registryUsernameHint, authConfig.Username)
+			registryDefaultOption = useDevSpaceRegistry
 		}
+	}
+	registryOptions := []string{useDockerHub, useDevSpaceRegistry, useOtherRegistry}
 
-		err = generated.SaveConfig(generatedConfig)
-		if err != nil {
-			log.Fatal(err)
-		}
+	selectedRegistry := survey.Question(&survey.QuestionOptions{
+		Question:     "Which registry do you want to use for storing your Docker images?",
+		DefaultValue: registryDefaultOption,
+		Options:      registryOptions,
+	})
+	var registryURL string
+
+	if selectedRegistry == useDockerHub {
+		registryURL = dockerHubHostname
+		registryLoginHint = fmt.Sprintf(registryLoginHint, "")
+	} else if selectedRegistry == useDevSpaceRegistry {
+		registryURL = cloudRegistryHostname
+		registryLoginHint = fmt.Sprintf(registryLoginHint, " "+cloudRegistryHostname)
+		cmd.useCloud = true
+	} else {
+		registryURL = survey.Question(&survey.QuestionOptions{
+			Question:     "Please enter the registry URL without image name:",
+			DefaultValue: "http://my.registry.tld/username",
+		})
+		registryURL = strings.Trim(registryURL, "/ ")
+		registryLoginHint = fmt.Sprintf(registryLoginHint, " "+registryURL)
+	}
+	log.WriteString("\n")
 
-		// Check if we should connect cluster
-		if connectCluster {
-			cmd.connectCluster()
+	if dockerClient != nil {
+		log.StartWait("Checking registry authentication")
+		authConfig, err := docker.Login(dockerClient, registryURL, "", "", true, false, false)
+		log.StopWait()
+
+		if err != nil || authConfig.Username == "" {
+			log.Fatalf("Registry authentication failed for %s.\n         %s", registryURL, registryLoginHint)
 		}
+	} else if selectedRegistry != useDevSpaceRegistry {
```

Comment:
Why check for useDevSpaceRegistry here? The user will get an error on provider login then since dockerClient cannot be created